### PR TITLE
Ppermute bug fix

### LINKED ISSUE #1703: ppermute is returning 0s
Tested with TPUs.

```python
@partial(jax.pmap, axis_name="shard")
def f(a):
  return jax.lax.ppermute(
      a, "shard", zip(range(8), [1, 0, 2, 3, 4, 5, 6, 7]))

a = np.ones((8, 2, 2))
print(f(a))
```

Prints out
```
[[[0. 0.]
  [0. 0.]]

 [[0. 0.]
  [0. 0.]]

 [[0. 0.]
  [0. 0.]]

 [[0. 0.]
  [0. 0.]]

 [[0. 0.]
  [0. 0.]]

 [[0. 0.]
  [0. 0.]]

 [[0. 0.]
  [0. 0.]]

 [[0. 0.]
  [0. 0.]]]
```

### PR DESCRIPTION
**Pull request by chaserileyroberts**
Fixes #1703

### CODE REVIEW

**Review by mattjj — APPROVED** [review]
Woo thank you!

File: tests/pmap_test.py [hunk 1]
```diff
@@ -331,6 +331,16 @@ def testCollectivePermuteCyclicGrad(self):
     expected = onp.roll(onp.pi + onp.arange(device_count), 1)
     self.assertAllClose(ans, expected, check_dtypes=False)
 
+  @jtu.skip_on_devices("cpu", "gpu")
+  def testIssue1703(self):
```
Comment (skye) [inline tests/pmap_test.py]: My favorite nit: rename to something more descriptive, like testPermuteWithGenerator

Comment (mattjj) [reply tests/pmap_test.py]: I'll tweak the name in a follow-up.

Comment (mattjj) [reply tests/pmap_test.py]: Done!

**Pull request merged**